### DISCUSSION TITLE
Supported embedded MSIL

### DISCUSSION THREAD

**Discussion by The-Futurist** [dotnet/csharplang#428] [General]
How about making it easier to embed IL instructions inline in C# methods/functions?

I know we have DynamicMethod but that doesn't have the capability to parse the IL at compile time or easily debug it at debug time.

Has anyone yet proposed this?

I suppose one could introduce a new keyword or something:

```
unmanaged
{
   LoadArg_0,
   Return
}

```
Alternatively one could look at the reasons we sometimes need to use DynamicMethod and introduce new C# language capabilities to support these unusual problems.

**Comment by jnm2** [comment]
Dupe of https://github.com/dotnet/csharplang/issues/226
